internal/resources/cluster/data_source_cluster.go

```diff
@@ -47,6 +47,9 @@ func dataSourceTMCClusterRead(ctx context.Context, d *schema.ResourceData, m int
 				_ = schema.RemoveFromState(d, m)
 				return false, nil
 			}
+			if clienterrors.IsUnauthorizedError(err) {
```

Comment:
there should be refresh here

Reply:
Yeah so instead of creating a new fresh util function, I'm trying to reuse the `setup()` function in `client.go` from authctx package, because essentially the setup() does two things:
- get a new auth token and set the value in `Client.Header` (which can be reused for refreshing tokens)
- set host value in `Client.Header`

And that's why I moved the second point out to where `setup()` are called, so that we don't set host value with redundancy. Feel free to let me know if a separate refresh util function is preferred.

Reply:
Okay so just made an update, and instead of reusing `setup()`, changed it to reusing `getUserAuthCtx()` for refreshing tokens